### PR TITLE
Minor updates towards the next release

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Update Parent POM
- [x] Update dependent libs
- [x] Rename "slave" to "agent" where it is compatible. No effective changes in EXE file names and service names since it may potentially break user system management scripts

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees done (response timeout)
@jenkinsci/code-reviewers WDYT?

File: src/main/resources/org/jenkinsci/modules/windows_slave_installer/Messages.properties [hunk 1]
```diff
@@ -23,8 +23,8 @@
 WindowsInstallerLink.DisplayName=Install as Windows Service
 WindowsInstallerLink.Description=Installs Jenkins as a Windows service to this system, so that Jenkins starts automatically when the machine boots.
 WindowsSlaveInstaller.ConfirmInstallation=\
-  This will install a slave agent as a Windows service, so that a Jenkins slave starts automatically when the machine boots. \
-  When the installation is over, this slave agent will exit to let the new service take over.
+  This will install a Jenkins agent as a Windows service, so that an agent starts automatically when the machine boots. \
+  When the installation is over, this agent agent will exit to let the new service take over.
```
Comment (oleg-nenashev) [inline src/main/resources/org/jenkinsci/modules/windows_slave_installer/Messages.properties]: agent agent

**Comment by slide** [comment]
Can we ship both jenkins-agent.exe AND jenkins-slave.exe for some period of time with jenkins-slave.exe deprecated for some period of time? The at some point we can get rid of jenkins-slave.exe.

**Comment by oleg-nenashev** [comment]
@slide 
We could ship new executable names and new config file names. So the new agents will be installed with these new names (though I am not a fan of such automatic service installation at all, I prefer manual WinSW setup and instalaltion). I doubt we can easily get rid of the existing ones after a period of time, but it's another story.

**Comment by oleg-nenashev** [comment]
Merging in order to unblock the WinSW update

**Pull request merged**